### ISSUE TITLE
Bug report  -  qqX specific 

### ISSUE THREAD

**Issue by Mark3xtrm** [TuxVinyards/qqX#5]
Even if link was ok, it fails to download Windows iso, and it only downloads the VirtIO drivers. At the end, in the VM's folder there's only the virtio-win.iso file speaking of iso files. I know I can download it manually, but obviously this is not something I'd like to do if I'm using a software like qqX. BTW: kudos, great piece of software

**Comment by TuxVinyards** [comment]
> "BTW: kudos, great piece of software."   

Thank you Mark for the praise. Appreciated.  Glad you are liking qqX.

> "Fail to download OS ISO (Mint)"

Thank you for flagging this.  If you want to download Mint, go ahead. Ignore the "bad link" message.  The downloader will work.  

I have traced the message warning to a problem where quickget has two homepages for  'linuxmint' .  The fix will be out in the new release, probably next week, if not before.  :+1: 

> "Even if link was ok, **_it_** fails to download Windows iso,"

It is Quickget that is the 'it' part in this case.  qqX only front ends quickget and for the most part can only report on what quickget is up to.  Also quickget  itself can only report on what Microsoft servers are doing. It can't control them.

> "EDIT: at this point, I tested this with Win 11, 10, and macOS, neither of which ISO's can be downloaded just using qqX"

It could be that you have a dynamic IP address that reset, or by the time you had tried again manually the Microsoft time-out block had lifted.   

qqX does give messages warning about these types of problems.

```428,39:       [[ $QGW_OS_Arr_NameSelected == "windows" ]] &&  printf "  Microsoft servers may not report accurately, may block repeat attempts or may need manual download ... " ```

If you think that this needs improved wording, let me know.

> "I know I can download it manually ...."

Yep. Absolutely. Automating this is already high on my agenda.  But your prompt has started me thinking about a possible solution.  See what happens in the next release ... :rocket: 

Also see <https://github.com/TuxVinyards/qqX/issues/4#issuecomment-1936961712>

**Comment by Mark3xtrm** [comment]
Yes, sorry, I misspoke: what I meant was that qqX does not download the ISO of the OS I want to virtualize, despite the link from qqX being reputed to be good.

Basically, when the downloads start, despite the fact that the links have been verified and the whole procedure seems to go well, in the end it downloads only the ISO of VirtIO, and in the folder there is therefore no The ISO of the OS I wanted to virtualize. There is to say, however, that actually in the messages, even though the Microsoft link was okay, it says that you were not able to download the ISO, and it is suggested that you do it manually.

Yes, as for any network problems, I do not deny that they could plague me. It's not that I have a bad connection, far from it, but the problem is that perhaps by having a non-denacted connection (I have a WiFi network with an LTE SIM), the Microsoft site breaks the boxes, and in fact even trying to download it manually tells me that it could not be done. If I restart the router and reload the ISO download page, sometimes the download is available

**Comment by TuxVinyards** [comment]
Have a look at the closed issue here:  <https://github.com/quickemu-project/quickemu/issues/898>  Might help.

**Comment by lj3954** [comment]
> Have a look at the closed issue here: [quickemu-project/quickemu#898](https://github.com/quickemu-project/quickemu/issues/898) Might help.

That is an irrelevant issue. Here, we're seeing quickget (or qqX) completely ignoring the macOS image. 

Microsoft blocking the download of a Windows ISO sometimes happens, often because of your IP address. This is not an issue that can be resolved. A macOS download should never fail, though. Please verify that you can replicate this issue, and provide the output. Also make sure you're using upstream quickget, issues you experience on forked versions should be reported to the fork's author instead.

**Comment by TuxVinyards** [comment]
The qqX LinuxMint issue has now been resolved. :+1: 

@Mark3xtrm  you might wish to file an issue at <https://github.com/quickemu-project/quickemu/issues> at this stage, if you want to discuss the Quickget/MacOS  issue with @lj3954 

If you are interested , there is currently MacOS development work going on, which @lj3954 has moved as quickget pull requests here: <https://github.com/quickemu-project/quickemu/pull/886>

Hopefully <https://github.com/quickemu-project/quickemu/issues/898> managed to address the Windows problems ....
